docs/gguf.md

```diff
@@ -0,0 +1,511 @@
+
+If a particular community key is widely used, it may be promoted to a standardized key.
+
+### General
+
+#### Required
+
+- **`general.architecture: string`**: describes what architecture this model implements. All lowercase ASCII, with only `[a-z0-9]+` characters allowed. Known values include:
+  - `llama`
+  - `mpt`
+  - `gptneox`
+  - `gptj`
+  - `gpt2`
+  - `bloom`
+  - `falcon`
+  - `rwkv`
+- **`general.quantization_version: uint32`**: version of quantization scheme. Not required if the model is not quantized (i.e. no tensors are quantized). If any tensors are quantized, this _must_ be present.
+- **`general.alignment: uint32`**: the global alignment to use, as described above. This can vary to allow for different alignment schemes, but it must be a multiple of 8.
+
+#### General metadata
+
+- `general.name`: The name of the model. This should be a human-readable name that can be used to identify the model. It should be unique within the community that the model is defined in.
+- `general.author`: The author of the model.
+- `general.url`: URL to the model's homepage. This can be a GitHub repo, a paper, etc.
+- `general.description: string`: free-form description of the model including anything that isn't covered by the other fields
+- `general.file_type: string`: type of the majority of the tensors in the file. This shouldn't have any semantic meaning and should be purely informational, hence the use of `string`.
+- `general.license: string`: SPDX license of the model
+
+#### Source metadata
+
+Information about where this model came from. This is useful for tracking the provenance of the model, and for finding the original source if the model is modified. For a model that was converted from GGML, for example, these keys would point to the model that was converted from.
+
+- `general.source.url: string`: URL to the source of the model. Can be a GitHub repo, a paper, etc.
+- `general.source.huggingface.repository: string`: Hugging Face model repository that this model is either hosted on or based on
+
+### LLM
+
+In the following, `[llm]` is used to fill in for the name of a specific LLM architecture. They will be used in each architecture's section.
+
+- `[llm].context_length: uint32`: Also known as `n_ctx`. length of the context (in tokens) that the model was trained on. For most architectures, this is the hard limit on the length of the input. Architectures, like RWKV, that are not reliant on transformer-style attention may be able to handle larger inputs, but this is not guaranteed.
+- `[llm].embedding_length: uint32`: Also known as `n_embd`. Embedding layer size.
+- `[llm].layer_count: uint32`: Also known as `n_layers`. The number of attention+feedforward layers (i.e. the bulk of the LLM). Does not include the input or embedding layers.
```

Comment:
This should probably change name to `[llm].block_count` since it signifies the number of _blocks_ of attention and feed-forward layers and not a direct count of layers.